### PR TITLE
[Tests][Spec Decode] Add gemma4 MTP acceptance rates test

### PR DESCRIPTION
**Pull request by TheEpicDolphin**
# This PR
Gemma4 MTP is lacking acceptance tests for both MRV1 and MRV2. I generalized the existing `test_dflash_acceptance_rates` test into `test_acceptance_rates` and extended it to test dflash (Qwen/Qwen3-8B + z-lab/Qwen3-8B-DFlash-b16) and gemma4 MTP (google/gemma-4-E4B-it + google/gemma-4-E4B-it-assistant). The test iterates over the specified datasets in `expected_acceptance_lengths`, runs the eval, and verifies that the acceptance length approximately matches what was expected. The test allows specifying a different set of datasets/acceptance lenghts per base/draft model pair.

Google didn't publish acceptance rates for the gemma4 draft models, so I just used the MRV1 implementation as a baseline. Fortunately, the MRV1 and MRV2 acceptance lengths match exactly across gsm8k, humaneval, and mt-bench.

While trying to enable these tests, I noticed and fixes these issues:
1. MRV1's Gemma4 MTP had a startup issue because #43957 added an embedding-dim equality guard to `_maybe_share_embeddings` in llm_base_proposer.py, but Gemma4 MTP intentionally shares dim-mismatched embeddings. Added a `_maybe_share_embeddings` override in gemma4.py.
2. Noticed that seq_lens weren't being updated with the num_rejected in `_prepare_decode_inputs_kernel` specifically in the gemma4 MTP case (`ADVANCE_DRAFT_POSITIONS = False`). Fixed by ensuring it the num_rejected subtraction is always accounted for. This made acceptance rates sub-par in MRV2.
3. In the case of probabilistic rejection sampling (using draft logits), the gumbel sample positions weren't being updated for the last N - 1 draft steps. Although KV positions shouldn't be updated, the gumbel sample positions should be.

# Benchmarks
Fixing #3 resulted in the following slight improvement in acceptance rates during probabilistic rejection sampling (with draft logits):
## GSM8K
| Metric | Run 2 | Run 1 | Δ |
|---|---|---|---|
| Successful / Failed requests | 1319 / 0 | 1319 / 0 | — |
| Maximum request concurrency | 64 | 64 | — |
| Benchmark duration (s) | 20.39 | 20.43 | +0.2% |
| Request throughput (req/s) | 64.70 | 64.56 | -0.2% |
| Output token throughput (tok/s) | 10181.39 | 10015.18 | -1.7% |
| Total token throughput (tok/s) | 58995.89 | 58728.88 | -0.5% |
| Mean TPOT (ms) | 5.23 | 5.10 | -2.5% |
| Median TPOT (ms) | 5.25 | 5.10 | -2.9% |
| P99 TPOT (ms) | 7.17 | 7.46 | +3.9% |
| Acceptance rate (%) | 74.22 | 74.46 | +0.24 pts |
| Acceptance length | 3.23 | 3.23 | flat |
| Drafts | 64,409 | 63,345 | -1.7% |
| Draft tokens | 193,227 | 190,035 | -1.7% |
| Accepted tokens | 143,405 | 141,506 | -1.3% |
| Per-position acceptance (%) — Pos 0 | 85.77 | 86.01 | +0.24 pts |
| Per-position acceptance (%) — Pos 1 | 73.83 | 74.08 | +0.25 pts |
| Per-position acceptance (%) — Pos 2 | 63.04 | 63.30 | +0.26 pts |

# Tests
```
(vllm) gdelfin@gb200-rack1-15:~/vllm$ python -m pytest "tests/v1/e2e/spec_decode/test_spec_decode.py::test_acceptance_rates" -q
Running 4 items in this shard
....                                                                                                                                                                          [100%]
================================================================================= warnings summary ==================================================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
  /home/gdelfin/vllm/.venv/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

tests/v1/e2e/spec_decode/test_spec_decode.py::test_acceptance_rates[False-dflash]
tests/v1/e2e/spec_decode/test_spec_decode.py::test_acceptance_rates[False-gemma4-mtp]
tests/v1/e2e/spec_decode/test_spec_decode.py::test_acceptance_rates[True-dflash]
tests/v1/e2e/spec_decode/test_spec_decode.py::test_acceptance_rates[True-gemma4-mtp]
  /home/gdelfin/.local/share/uv/python/cpython-3.12.13-linux-aarch64-gnu/lib/python3.12/multiprocessing/popen_fork.py:66: DeprecationWarning: This process (pid=824917) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

tests/v1/e2e/spec_decode/test_spec_decode.py: 12 warnings
  /home/gdelfin/vllm/.venv/lib/python3.12/site-packages/datasets/utils/_dill.py:385: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    obj.co_lnotab,  # for < python 3.10 [not counted in args]

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
4 passed, 32 warnings in 1757.26s (0:29:17)
```

### CODE REVIEW

File: vllm/v1/worker/gpu/spec_decode/autoregressive/speculator.py [hunk 1]
```diff
@@ -630,19 +632,18 @@ def _prepare_decode_inputs_kernel(
...
         position = tl.load(positions_ptr + req_idx)
         position = tl.minimum(position + 1, max_model_len - 1)
         tl.store(positions_ptr + req_idx, position)
-
-        target_seq_len = tl.load(target_seq_lens_ptr + req_idx)
-        num_rejected = tl.load(num_rejected_ptr + req_idx)
-        seq_len = target_seq_len - num_rejected
         seq_len = tl.minimum(seq_len + 1, max_model_len)
-        tl.store(seq_lens_ptr + req_idx, seq_len)
+    tl.store(seq_lens_ptr + req_idx, seq_len)
```
Comment (TheEpicDolphin) [inline vllm/v1/worker/gpu/spec_decode/autoregressive/speculator.py:650]: `seq_len` should account for `num_rejected` regardless of whether we are advancing draft positions or not.
